cmd/utils/flags.go

```diff
@@ -1127,7 +1131,12 @@ func setTxFeeRecipient(ctx *cli.Context, ks *keystore.KeyStore, cfg *eth.Config)
 			Fatalf("`etherbase` and `tx-fee-recipient` flag should not be used together. `miner.validator` and `tx-fee-recipient` constitute both of `etherbase`' functions")
 		}
 		txFeeRecipient = ctx.GlobalString(TxFeeRecipientFlag.Name)
+
+		if !ctx.GlobalIsSet(MinerValidatorFlag.Name) {
+			Fatalf("`miner.validator` flag should be set if `tx-fee-recipient` is set")
```

Comment:
i don't agree.. if you are a full node, you might set the tx-fee-recipient for the gateway fee, and you're not a miner, so you don't set miner.validator, why is thi s needed?

Reply:
True, missed those cases